### ISSUE TITLE
make_DBscores is failing: Error in vapply(all_objects, nrow, integer(1L), USE.NAMES = FALSE)

### ISSUE THREAD

**Issue by smped** [bjmt/universalmotif#29]
Hi,

Thanks for a super-useful package. I'm having a bit of trouble running `make_DBscores()` unfortunately.

The steps I went through are

1. I obtained the set of HOCOMOCv11 motifs in MEME format from https://hocomoco11.autosome.org/final_bundle/hocomoco11/core/HUMAN/mono/HOCOMOCOv11_core_HUMAN_mono_meme_format.meme
2. These were loaded into an `R` session using the following code:

``` r
library(universalmotif)
> motifs <- here::here("HOCOMOCOv11_core_HUMAN_mono_meme_format.meme") |>
    read_meme() |>
    to_df()
```

Trying to make my own P-value database then fails. I also tested using multiple methods together and "EUCL" by itself. The same failure message was observed.

``` r
db <- make_DBscores(to_list(motifs), method = "PCC", nthreads = 4)
```

```
 *** Total runtime: 5.125894 mins ***
Error in vapply(all_objects, nrow, integer(1L), USE.NAMES = FALSE) : 
  values must be length 1,
 but FUN(X[[2]]) result is length 0
```

SessionInfo()

```
R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C               LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8    LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Adelaide
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] universalmotif_1.22.2

loaded via a namespace (and not attached):
 [1] gtable_0.3.5            jsonlite_1.8.8          dplyr_1.1.4             compiler_4.4.0          crayon_1.5.3           
 [6] tidyselect_1.2.1        Rcpp_1.0.13             Biostrings_2.72.1       scales_1.3.0            IRanges_2.38.1         
[11] yaml_2.3.9              here_1.0.1              ggplot2_3.5.1           R6_2.5.1                XVector_0.44.0         
[16] generics_0.1.3          GenomeInfoDb_1.40.1     BiocGenerics_0.50.0     MASS_7.3-61             tibble_3.2.1           
[21] MatrixGenerics_1.16.0   munsell_0.5.1           rprojroot_2.0.4         GenomeInfoDbData_1.2.12 pillar_1.9.0           
[26] rlang_1.1.4             utf8_1.2.4              cli_3.6.3               magrittr_2.0.3          zlibbioc_1.50.0        
[31] grid_4.4.0              rstudioapi_0.16.0       lifecycle_1.0.4         S4Vectors_0.42.1        vctrs_0.6.5            
[36] glue_1.7.0              stats4_4.4.0            colorspace_2.1-1        fansi_1.0.6             httr_1.4.7             
[41] matrixStats_1.3.0       tools_4.4.0             pkgconfig_2.0.3         UCSC.utils_1.0.0 
```

**Comment by bjmt** [comment]
Hi Stevie,

Thanks for you kind words about the package! Glad it's of use to you.

I appreciate the detailed report, it was easy for me to recreate your error. I'm pretty sure I've identified what's wrong, it seems like a weirdly obvious error. Not sure how I haven't encountered it before. In any case, I've pushed a fix to the devel branch on Bioconductor (v1.23.7). If all the checks work out on the Bioconductor build machines, then I'll push the fix to the release version as well.

Thanks for reporting this.

Ben
